nagios.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// NagiosData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func NagiosData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return nagiosData(func(project, pod string) *exec.Cmd {
+		return exec.Command("oc", "exec", pod, "cat", "/var/log/nagios/status.dat")
+	}, project, outFor, errOutFor)
+}
+
+// A getNagiosDataCmdFactory generates commands to get the raw nagios status data
+// in the provided project from the provided pod
+type getNagiosDataCmdFactory func(project, pod string) *exec.Cmd
+
+// nagiosData will dump a JSON representation of the current nagios status into
+// the io.Writer returned from outFor and log any errors it encounters doing so
+// into the io.Writer returned from errOutFor.
+func nagiosData(cmdFactory getNagiosDataCmdFactory, project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return func() error {
+		stdout, stdoutCloser, err := outFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stdoutCloser.Close()
+
+		stderr, stderrCloser, err := errOutFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stderrCloser.Close()
+
+		var stdoutBuff bytes.Buffer
+
+		nagiosPod, err := getProjectPodNamesBySubstr(project, "nagios")
+		if err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		if len(nagiosPod) == 0 {
+			err := errors.New("could not find a nagios pod in the project '" + project + "'")
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		cmd := cmdFactory(project, nagiosPod[0])
+
+		if err := runCmdCaptureOutput(cmd, &stdoutBuff, stderr); err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		jsonStatus := parseNagiosStatus(string(stdoutBuff.Bytes()))
+		stdout.Write(jsonStatus)
+
+		return nil
```

Comment:
@philbrookes why we just not do only one return err; and not doing it to different cases and if err happens we pass it to caller function?